### PR TITLE
Set current slug from jupyter extensions

### PR DESCRIPTION
**Pull request by PrajwalPrashanth**
* [x] slug helper file
* [x] set current slug from nb extension
* [x] Discuss about not directly calling `set_current_slug` from extension instead of having it in `import jovian`
* [x] Implement in lab
* [x] Write test for `utils/slug.py`
* [x] Discuss the edge case
* [x] It wont handle .jovianrc deleted case

Edge Case: If user commits a nb named Untitled , changes it filename, creates a new nb again which will be named Untitled by default and when he commits that it wont be a new notebook. (Solution could be delete duplicate slug when a same is updated with a new key)

### CODE REVIEW

**Comment by PrajwalPrashanth** [comment]
This is ready for review as well @aakashns . Tests are remaining as i'm facing issues there, will update them after talking to rohit.

**Comment by PrajwalPrashanth** [comment]
@aakashns removed this from WIP, tests are added.

File: jovian/utils/commit.py [hunk 1]
```diff
@@ -52,37 +53,38 @@ def commit(message=None,
...
     .. attention::
-        Pass notebook's name to `filename` argument, in certain environments like Jupyter Lab and password protected 
+        Pass notebook's name to `filename` argument, in certain environments like Jupyter Lab and password protected
         notebooks sometimes it may fail to detect notebook automatically.
     .. |Jovian| raw:: html
 
         <a href="https://jovian.ml/?utm_source=docs" target="_blank"> Jovian.ml </a>
     """
     global _current_slug
+    _current_slug = get_current_slug()
```
Comment (aakashns) [inline jovian/utils/commit.py:87]: This causes duplication of the value of `_current_slug` in 2 files: slug.py and commit.py. This can cause problems, as the two values may go out of sync. 

Avoid this duplication and have a single source of truth.

File: jovian/tests/utils/test_slug.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+@mock.patch("jovian.utils.slug._current_slug", "test_slug")
+def test_get_current_slug():
+    assert get_current_slug() == "test_slug"
+
+
+@mock.patch("jovian.utils.slug.get_notebook_slug", return_value="test_slug")
+def test_set_current_slug(mock_get_notebook_slug):
+    set_current_slug("fake_filename")
+    assert slug._current_slug == "test_slug"
```
Comment (aakashns) [inline jovian/tests/utils/test_slug.py:14]: This test can be improved. We want to test that `set_current_slug` (ideally it should be called `detect_current_slug`) gets the slug for the given `filename` from the `.jovianrc` file. However, by mocking `get_notebook_slug`, we are completely ignoring the `filename` argument. 

Try mocking the `rcfile` instead of mocking `get_notebook_slug`, or mock `get_notebook_slug` so that it returns `test_slug` only if the argument is `fake_filename`

File: jovian/utils/slug.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from jovian.utils.rcfile import get_notebook_slug
+
+_current_slug = None
+
+
+def set_current_slug(filename):
```
Comment (aakashns) [inline jovian/utils/slug.py:6]: This method name is a bit misleading. One would expect that `set_current_slug` will set the `_current_slug` to the given argument.

Might be better to call this `detect_current_slug`

**Comment by aakashns** [comment]
After reviewing, I felt the right place for the cached value of `_current_slug` is in `rcfile.py`. Also, we can do the caching seamlessly using `get_notebook_slug` and `set_notebook_slug` to avoid the cached value going out of sync with the `.jovianrc` file
